File: day21_graph_traversal/bfs.go
package main


//Node type holds the data required fro DFS
type Node struct {

	name string
	children []*Node

}

// BFS traverse along the graph keeping a depth first approach
func (node *Node) BFS(array []string) []string {
	// append the node which we touched first
	queue:=[]*Node{node}

	for len(queue)>0 {
		// store queue front value in current
		current:=queue[0]
		// remove front value from queue
		queue=queue[1:]
		array=append(array, current.name)
		for _, childNode := range current.children {
			queue = append(queue, childNode)
		}
	}

	return array
	
}

File: day21_graph_traversal/bfs.cpp
#include <vector>
#include <iostream>
#include <string>
#include<deque>

using namespace std;

class Node {

    public:

    string name;
    vector <Node *> children;

    Node(string name){
        this->name= name;
    }

    vector<string> BFS(vector<string> * array){
        deque<Node* > queue{this};

        int counter=0;
        while(!queue.empty()){
            Node  current = *queue.front();
            queue.pop_front();
            array->push_back(current.name);
            for (int i = 0; i < current.children.size(); i++)
            {
                queue.push_back(current.children[i]);
            }
            
        }
        return *array;
    }  
};